src/main/java/ch/njol/yggdrasil/DefaultYggdrasilInputStream.java

```diff
@@ -283,20 +256,16 @@ protected Class<?> readClass() throws IOException {
 			default:
 				throw new YggdrasilException("Internal error; " + type);
 		}
-		while (dim-- > 0)
-			c = Array.newInstance(c, 0).getClass();
-		return c;
+		while (dimensions-- > 0)
```

Comment:
This seems quite yucky, I wonder if there's a more efficient way that doesn't involve repeated array creation by reflection.

Reply:
Newer Java versions offer [Class#arrayType()](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Class.html#arrayType()), unfortunately there's no better way to do this with Java 8.

Reply:
This is already in the CollectionUtils class, `CollectionUtils.arrayType(Class)`